test/contracts/promises/promises-common/src/lib.rs

```diff
@@ -0,0 +1,85 @@
+#![no_std]
+
+pub use elrond_wasm::{Address, Vec};
+
+pub const PARENT_ADDRESS: [u8; 32] = [
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0F, 0x0F, b'p', b'a', b'r', b'e', b'n', b't',
+    b'S', b'C', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.',
+];
+pub const CHILD_ADDRESS: [u8; 32] = [
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0F, 0x0F, b'c', b'h', b'i', b'l', b'd', b'S',
+    b'C', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.',
+];
+
+pub const FIRST_CONTRACT_ADDRESS: [u8; 32] = [
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0F, 0x0F, b'f', b'i', b'r', b's', b't', b'C',
+    b'o', b'n', b't', b'r', b'a', b'c', b't', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.',
+];
+pub const SECOND_CONTRACT_ADDRESS: [u8; 32] = [
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0F, 0x0F, b's', b'e', b'c', b'o', b'n', b'd',
+    b'C',b'o', b'n', b't', b'r', b'a', b'c', b't', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.',
+];
+pub const THIRD_CONTRACT_ADDRESS: [u8; 32] = [
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x0F, 0x0F, b't', b'h', b'i', b'r', b'd', b'C',
+    b'o', b'n', b't', b'r', b'a', b'c', b't', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.', b'.',
+];
+
+pub const GAS_100K: i64 = 100000;
+pub const GAS_500K: i64 = 500000;
+pub const GAS_5M: i64 = 5000000;
+pub const GAS_10M: i64 = 10000000;
+pub const GAS_50M: i64 = 50000000;
+pub const GAS_100M: i64 = 100000000;
+
+pub const ZERO: [u8; 32] = [0u8; 32];
+pub const EMPTY_SLICE: &[u8] = &[];
+
+pub fn construct_storage_key(key_parts: &[&[u8]]) -> Vec<u8> {
```

Comment:
The implementation is good, just putting it out here that you can also use 
`BoxedBytes::from_concat(key_parts: &[&[u8]])`,
which already does the same. The result is a BoxedBytes object instead of Vec, but you can also call `.as_slice()` on it.

Reply:
I need to start writing all the test contracts in Rust.